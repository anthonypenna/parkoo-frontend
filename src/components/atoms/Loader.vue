<template>
  <Fade>
    <div class="loader">
      <div class="loader__ball loader__ball--one"></div>
      <div class="loader__ball loader__ball--two"></div>
      <div class="loader__ball loader__ball--three"></div>
    </div>
  </Fade>
</template>

<script lang="ts">
import Vue from "vue";
import Fade from "@/components/atoms/Fade.vue";

export default Vue.extend({
  name: "Loader",
  components: { Fade },
});
</script>

<style lang="scss" scoped>
.loader {
  display: flex;
  justify-content: center;

  &__ball {
    width: 16px;
    height: 16px;
    margin: 0 8px;
    background: #ff9979;
    border-radius: 50%;
  }

  &__ball--one {
    animation: jump 1s ease-in-out 0s infinite;
  }

  &__ball--two {
    animation: jump 1s ease-in-out 0.1s infinite;
  }

  &__ball--three {
    animation: jump 2s ease-in-out 0.2s infinite;
  }
}

@keyframes jump {
  50% {
    opacity: 0.2;
    transform: scale(0.5) translateY(16px);
  }
}
</style>
