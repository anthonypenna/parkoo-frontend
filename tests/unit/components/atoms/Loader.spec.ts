import { shallowMount } from '@vue/test-utils'
import Loader from '@/components/atoms/Loader.vue'

describe('<Loader />', () => {
  it('should render correctly', () => {
    const wrapper = shallowMount(Loader)
    expect(wrapper.html()).toMatchSnapshot()
  })
})
