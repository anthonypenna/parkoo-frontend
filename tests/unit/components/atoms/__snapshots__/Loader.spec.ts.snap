// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader /> should render correctly 1`] = `
<fade-stub>
  <div class="loader">
    <div class="loader__ball loader__ball--one"></div>
    <div class="loader__ball loader__ball--two"></div>
    <div class="loader__ball loader__ball--three"></div>
  </div>
</fade-stub>
`;
